### PR TITLE
Option to forcefully reserve sequence numbers

### PR DESCRIPTION
**Pull request by MGathier**
Reserve sequence numbers without checking (used in enterprise edition).

If creation of index should fail for any reason, do not complete the segment yet.

### CODE REVIEW

File: axonserver/src/main/java/io/axoniq/axonserver/localstorage/EventStorageEngine.java [hunk 1]
```diff
@@ -88,8 +88,9 @@ default void close() {
     /**
      * Reserve the sequence numbers of the aggregates in the provided list to avoid collisions during store.
      * @param events list of events to store
+     * @param force force the specified sequence numbers to be correct
      */
-    default void reserveSequenceNumbers(List<SerializedEvent> events) {
+    default void reserveSequenceNumbers(List<SerializedEvent> events, boolean force) {
```
Comment (m1l4n54v1c) [inline axonserver/src/main/java/io/axoniq/axonserver/localstorage/EventStorageEngine.java]: Value of this parameter is always set to `false`. Do we need it?

File: axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/Synchronizer.java [hunk 2]
```diff
@@ -85,7 +85,11 @@ private void syncAndCloseFile() {
         try {
             WritePosition toSync = syncAndCloseFile.pollFirst();
             if (toSync != null) {
-                closeFile(toSync);
+                try {
+                    closeFile(toSync);
+                } catch (Exception ex) {
+                    syncAndCloseFile.add(toSync);
```
Comment (m1l4n54v1c) [inline axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/Synchronizer.java]: Should we log `ex`?

**Comment by m1l4n54v1c** [comment]
Ah, I now see a comment that this parameter might be used in EE. I'll reformulate my question: "In which situations we want to force reserving of 'unexpected' sequence numbers?"

File: axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/Synchronizer.java (hunk 2 above)
Comment (MGathier) [reply axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/Synchronizer.java]: Good catch, looked at the exception handling again and the exception code was never reached as closeFile already caught the exception. updated the error handling to make this the only place where exceptions are caught.

**Pull request merged**